krita_batch_exporter/Infrastructure.py

```diff
@@ -14,8 +14,46 @@
 
 
 def dataToPIL(wnode):
-    img = wnode.node.projectionPixelData(*wnode.bounds).data()
-    img = Image.frombytes("RGBA", wnode.size, img, "raw", "BGRA", 0, 1)
+    """
+    Returns an Image object using the Pillow library with 8-bit sRGB data that Pillow can then process.
+    If the input node's color space isn't 8-bit sRGB, uses Krita to convert the document to the right color space.
+    """
+
+    SRGB_PROFILE = "sRGB-elle-V2-srgbtrc.icc"
+
+    [x, y, w, h] = wnode.bounds
+
+    if wnode.node.colorModel() == "RGBA" and wnode.node.colorDepth() == "U8" and wnode.node.colorProfile() == SRGB_PROFILE:
+        pixel_data = wnode.node.projectionPixelData(x, y, w, h).data()
+        img = Image.frombytes("RGBA", wnode.size, pixel_data, "raw", "BGRA", 0, 1)
+        return img
+        
+    # Create a temporary Document for color space conversion. This document will have
+    # one transparent layer,
```

Comment:
Could you please remove the comments? Now the info is in the docstring

Reply:
I think it's important to make it clear that the document created by `Krita::createDocument` will have one transparent layer created for it, as it's accessed with `temp_doc.topLevelNodes()[0]` below. This information is available in Krita's docs but it isn't very easy to navigate.

I will remove it if you insist, but I think it makes the code *less* future-proof, imho.

Reply:
I was saying that for comments in general, not this one specifically. E.g. `KI.createDocument` implies that you're creating a document.

Reply:
Fine. I'd really appreciate it if you read the whole reply, but you get what you want.